vault-plugins/flant_iam/model/group.go

```diff
@@ -420,3 +420,46 @@ func (r *GroupRepository) FindAllParentGroupsForServiceAccountUUID(tenantUUID Te
 	}
 	return r.findAllParentGroupsForGroupUUIDs(tenantUUID, groups)
 }
+
+func (r *GroupRepository) FindAllParentGroupsForGroupUUID(tenantUUID TenantUUID, groupUUID GroupUUID) (map[GroupUUID]struct{}, error) {
+	return r.findAllParentGroupsForGroupUUIDs(tenantUUID, map[GroupUUID]struct{}{groupUUID: {}})
+}
+
+func (r *GroupRepository) FindAllSubjectsFor(tenantUUID TenantUUID, users []UserUUID, serviceAccounts []ServiceAccountUUID, groups []GroupUUID) (map[UserUUID]struct{}, map[ServiceAccountUUID]struct{}, error) {
+	resultUsers := make(map[UserUUID]struct{})
+	resultSAs := make(map[ServiceAccountUUID]struct{})
+
+	for _, user := range users {
+		resultUsers[user] = struct{}{}
+	}
+	for _, sa := range serviceAccounts {
+		resultSAs[sa] = struct{}{}
+	}
+
+	for _, groupUUID := range groups {
+		groupsUsers, groupsSAs, err := r.FindAllSubjectsForGroupUUID(tenantUUID, groupUUID)
+		if err != nil {
+			return nil, nil, err
+		}
+		for user := range groupsUsers {
+			resultUsers[user] = struct{}{}
+		}
+		for sa := range groupsSAs {
+			resultSAs[sa] = struct{}{}
+		}
+	}
+
+	return resultUsers, resultSAs, nil
+}
```

Comment:
Just a thought. Looking at this PR, I think this code would benefit a lot from Set type. it would make code more clean and eliminate helpers for slices, maps, and uuids merging.

```go
type Set map[string]struct{} 
// Add(els ...string)
resultUsers.Add(groupUsers...)
resultSAs.Add(groupsSAs...)


// Slice() []string; 
```

Reply:
If we will user Set  Type - we will lose information about Set content in method signature